src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -116,6 +116,7 @@ public final class WorkflowJob extends Job<WorkflowJob,WorkflowRun> implements L
     private transient LazyBuildMixIn<WorkflowJob,WorkflowRun> buildMixIn;
     /** @deprecated replaced by {@link DisableConcurrentBuildsJobProperty} */
     private @CheckForNull Boolean concurrentBuild;
+    private Boolean disableRestartFromStage;
```

Comment:
"Restart from stage" is a Declarative-specific concept, so we do not want to add this kind of option to this plugin, which is more generic. I would probably have https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/560 define a `class DisableRestartFromStage extends InvisibleAction` and then have `Utils.updateJobProperties` add that action to the job when `disableRestartFromStage(true)` is used (or remove it if someone uses `disableRestartFromStage(false)`) instead of setting this field, and then have `RestartDeclarativePipelineAction.isRestartableFromStage` check for the existence of that action.

Reply:
@dwnusbaum much appreciated for the feedback. i will be looking into this 🙏 please bare with me, as i am contributing to Jenkins for the first time, so i am getting my head around the entire codebase

Reply:
No worries, let me know if you have other questions. I think that https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/560 along with the changes I described above would be a good approach.

Reply:
Sounds good, will do, thanks 🙏